lib/utils/extensions/iterable_extension.dart

```diff
@@ -0,0 +1,14 @@
+/// Extension on Iterable
+extension IterableExtension<T> on Iterable<T> {
+  /// Returns the first element in the iterable that satisfies the given [test]
+  /// function, or `null` if no element satisfies the condition.
+  T? firstWhereOrNull(bool Function(T) test) {
+    for (final element in this) {
+      if (test(element)) {
+        return element;
+      }
+    }
+
+    return null;
+  }
+}
```

Comment:
https://pub.dev/documentation/collection/latest/collection/IterableExtension/firstWhereOrNull.html

```suggestion
import 'package:collections/collections.dart';
```

Reply:
But I want to avoid using packages in our package

Reply:
`collections` is a common enough package, I don't think it makes sense to avoid it. Let's add it here and remove the custom implementation.

My reasoning is -- the package is tried and tested, while maintaining a set of utilities like this incurs overhead. Ideally I'd want to see some unit-tests for this function, the tests and the function itself have to be maintained etc.

Why do all that when we have a ready-made solution in a package. Managing dependencies is a simpler task than what I described above.